### ISSUE TITLE
abort() in Regex::searchAll due to out of range error is leading to nginx worker process segfault

### ISSUE THREAD

**Issue by defanator** [SpiderLabs/ModSecurity#1304]
A couple of synthetic requests generated by `nikto` tool lead to nginx worker process segfault.

Request 1:

    curl -D - "http://localhost/vb/ajax/api/hook/decodeArguments?arguments=O%3A12%3A%22vB_dB_Result%22%3A2%3A%7Bs%3A5%3A%22%00%2A%00db%22%3BO%3A17%3A%22vB_Database_MySQL%22%3A1%3A%7Bs%3A9%3A%22functions%22%3Ba%3A1%3A%7Bs%3A11%3A%22free_result%22%3Bs%3A6%3A%22assert%22%3B%7D%7Ds%3A12%3A%22%00%2A%00recordset%22%3Bs%3A25%3A%22system%28%27cat%20%2Fetc%2Fpasswd%27%29%22%3B%7D"

error log:

    2017/01/12 10:40:44 [info] 25781#25781: *180187 [client 127.0.0.1] ModSecurity: Warning. Matched "Operator `@validatebyterange' with parameter `1-255' against variable `REQUEST_URI' (Value: `/vb/ajax/api/hook/decodeArguments?arguments=O:12:"vB_dB_Result":2:{s:5:"\x00*\x00db";O:17:"vB_Databa (128 characters omitted)' ) " at REQUEST_URI [file "/etc/nginx/modsec/owasp-v3/rules/REQUEST-920-PROTOCOL-ENFORCEMENT.conf"] [line "490"] [id "920270"] [rev "2"] [msg "Invalid character in request (null character)"] [data ""] [severity "0"] [ver "OWASP_CRS/3.0.0"] [maturity "9"] [accuracy "9"] [hostname "127.0.0.1"] [uri "/vb/ajax/api/hook/decodeArguments"] [unique_id "148421764416.039730"], client: 127.0.0.1, server: , request: "GET /vb/ajax/api/hook/decodeArguments?arguments=O%3A12%3A%22vB_dB_Result%22%3A2%3A%7Bs%3A5%3A%22%00%2A%00db%22%3BO%3A17%3A%22vB_Database_MySQL%22%3A1%3A%7Bs%3A9%3A%22functions%22%3Ba%3A1%3A%7Bs%3A11%3A%22free_result%22%3Bs%3A6%3A%22assert%22%3B%7D%7Ds%3A12%3A%22%00%2A%00recordset%22%3Bs%3A25%3A%22system%28%27cat%20%2Fetc%2Fpasswd%27%29%22%3B%7D HTTP/1.1", host: "localhost"
    2017/01/12 10:40:44 [info] 25781#25781: *180187 [client 127.0.0.1] ModSecurity: Warning. Matched "Operator `@validatebyterange' with parameter `1-255' against variable `ARGS:arguments' (Value: `O:12:"vB_dB_Result":2:{s:5:"\x00*\x00db";O:17:"vB_Database_MySQL":1:{s:9:"functions";a:1:{s:11:"free (84 characters omitted)' ) " at ARGS:arguments [file "/etc/nginx/modsec/owasp-v3/rules/REQUEST-920-PROTOCOL-ENFORCEMENT.conf"] [line "490"] [id "920270"] [rev "2"] [msg "Invalid character in request (null character)"] [data ""] [severity "0"] [ver "OWASP_CRS/3.0.0"] [maturity "9"] [accuracy "9"] [hostname "127.0.0.1"] [uri "/vb/ajax/api/hook/decodeArguments"] [unique_id "148421764416.039730"], client: 127.0.0.1, server: , request: "GET /vb/ajax/api/hook/decodeArguments?arguments=O%3A12%3A%22vB_dB_Result%22%3A2%3A%7Bs%3A5%3A%22%00%2A%00db%22%3BO%3A17%3A%22vB_Database_MySQL%22%3A1%3A%7Bs%3A9%3A%22functions%22%3Ba%3A1%3A%7Bs%3A11%3A%22free_result%22%3Bs%3A6%3A%22assert%22%3B%7D%7Ds%3A12%3A%22%00%2A%00recordset%22%3Bs%3A25%3A%22system%28%27cat%20%2Fetc%2Fpasswd%27%29%22%3B%7D HTTP/1.1", host: "localhost"
    2017/01/12 10:40:44 [info] 25781#25781: *180187 [client 127.0.0.1] ModSecurity: Warning. Matched "Operator `@pmf' with parameter `unix-shell.data' against variable `ARGS:arguments' (Value: `O:12:"vB_dB_Result":2:{s:5:"\x00*\x00db";O:17:"vB_Database_MySQL":1:{s:9:"functions";a:1:{s:11:"free (84 characters omitted)' ) " at ARGS:arguments [file "/etc/nginx/modsec/owasp-v3/rules/REQUEST-932-APPLICATION-ATTACK-RCE.conf"] [line "412"] [id "932160"] [rev "1"] [msg "Remote Command Execution: Unix Shell Code Found"] [data ""] [severity "0"] [ver "OWASP_CRS/3.0.0"] [maturity "1"] [accuracy "8"] [hostname "127.0.0.1"] [uri "/vb/ajax/api/hook/decodeArguments"] [unique_id "148421764416.039730"], client: 127.0.0.1, server: , request: "GET /vb/ajax/api/hook/decodeArguments?arguments=O%3A12%3A%22vB_dB_Result%22%3A2%3A%7Bs%3A5%3A%22%00%2A%00db%22%3BO%3A17%3A%22vB_Database_MySQL%22%3A1%3A%7Bs%3A9%3A%22functions%22%3Ba%3A1%3A%7Bs%3A11%3A%22free_result%22%3Bs%3A6%3A%22assert%22%3B%7D%7Ds%3A12%3A%22%00%2A%00recordset%22%3Bs%3A25%3A%22system%28%27cat%20%2Fetc%2Fpasswd%27%29%22%3B%7D HTTP/1.1", host: "localhost"
    terminate called after throwing an instance of 'std::out_of_range'
      what():  basic_string::basic_string: __pos (which is 144) > this->size() (which is 28)
    2017/01/12 10:40:45 [notice] 25780#25780: signal 17 (SIGCHLD) received
    2017/01/12 10:40:45 [alert] 25780#25780: worker process 25781 exited on signal 6 (core dumped)

backtrace:

    (gdb) bt
    #0  0x00007f18dc6757ef in raise () from /lib/x86_64-linux-gnu/libc.so.6
    #1  0x00007f18dc6773ea in abort () from /lib/x86_64-linux-gnu/libc.so.6
    #2  0x00007f18db13f58d in __gnu_cxx::__verbose_terminate_handler () at ../../../../src/libstdc++-v3/libsupc++/vterminate.cc:95
    #3  0x00007f18db13d336 in __cxxabiv1::__terminate (handler=<optimized out>) at ../../../../src/libstdc++-v3/libsupc++/eh_terminate.cc:47
    #4  0x00007f18db13d381 in std::terminate () at ../../../../src/libstdc++-v3/libsupc++/eh_terminate.cc:57
    #5  0x00007f18db13d599 in __cxxabiv1::__cxa_throw (obj=obj@entry=0x55ad5255f3f0, tinfo=0x7f18db428a98 <typeinfo for std::out_of_range>, dest=0x7f18db153000 <std::out_of_range::~out_of_range()>)
        at ../../../../src/libstdc++-v3/libsupc++/eh_throw.cc:87
    #6  0x00007f18db166ff7 in std::__throw_out_of_range_fmt (__fmt=__fmt@entry=0x7f18dc1c6460 "%s: __pos (which is %zu) > this->size() (which is %zu)")
        at ../../../../../src/libstdc++-v3/src/c++11/functexcept.cc:101
    #7  0x00007f18dc1b4168 in std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_check (__s=0x7f18dc1c6443 "basic_string::basic_string", __pos=144, this=0x7fffc2491f00)
        at /usr/include/c++/6/bits/basic_string.h:262
    #8  std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::basic_string (__n=25, __pos=144, __str="O:12:\"vB_dB_Result\":2:{s:5:\"", this=0x7fffc2491f20)
        at /usr/include/c++/6/bits/basic_string.h:415
    #9  modsecurity::Utils::Regex::searchAll (this=<optimized out>, s=...) at utils/regex.cc:89
    #10 0x00007f18dc1a7028 in modsecurity::operators::Rx::evaluate (this=<optimized out>, transaction=0x55ad51f3c670, rule=0x55ad4a93e810, input=...) at operators/rx.cc:39
    #11 0x00007f18dc19b73a in modsecurity::operators::Operator::evaluateInternal (this=0x55ad4a92f350, transaction=transaction@entry=0x55ad51f3c670, rule=rule@entry=0x55ad4a93e810, 
        a="O:12:\"vB_dB_Result\":2:{s:5:\"\000*\000db\";O:17:\"vB_Database_MySQL\":1:{s:9:\"functions\";a:1:{s:11:\"free_result\";s:6:\"assert\";}}s:12:\"\000*\000recordset\";s:25:\"system('cat /etc/passwd')\";}")
        at operators/operator.cc:80
    #12 0x00007f18dc161961 in modsecurity::Rule::executeOperatorAt (this=this@entry=0x55ad4a93e810, trasn=trasn@entry=0x55ad51f3c670, key="ARGS:arguments", 
        value="O:12:\"vB_dB_Result\":2:{s:5:\"\000*\000db\";O:17:\"vB_Database_MySQL\":1:{s:9:\"functions\";a:1:{s:11:\"free_result\";s:6:\"assert\";}}s:12:\"\000*\000recordset\";s:25:\"system('cat /etc/passwd')\";}") at rule.cc:270
    #13 0x00007f18dc166f62 in modsecurity::Rule::evaluate (this=this@entry=0x55ad4a93e810, trasn=trasn@entry=0x55ad51f3c670) at rule.cc:597
    #14 0x00007f18dc15c64b in modsecurity::Rules::evaluate (this=0x55ad49a626f0, phase=phase@entry=3, transaction=transaction@entry=0x55ad51f3c670) at rules.cc:231
    #15 0x00007f18dc14eea3 in modsecurity::Transaction::processRequestBody (this=0x55ad51f3c670) at transaction.cc:792
    #16 0x00007f18dc150195 in modsecurity::msc_process_request_body (transaction=<optimized out>) at transaction.cc:1779
    #17 0x00007f18dc43cb7e in ngx_http_modsecurity_pre_access_handler (r=0x55ad52139170) at ../ModSecurity-nginx//src/ngx_http_modsecurity_pre_access.c:199
    #18 0x000055ad4828821f in ngx_http_core_generic_phase ()
    #19 0x000055ad482838b5 in ngx_http_core_run_phases ()
    #20 0x000055ad4828ea05 in ngx_http_process_request ()
    ...

Request 2:

    curl -D - "http://localhost/vbulletin/ajax/api/hook/decodeArguments?arguments=O%3A12%3A%22vB_dB_Result%22%3A2%3A%7Bs%3A5%3A%22%00%2A%00db%22%3BO%3A17%3A%22vB_Database_MySQL%22%3A1%3A%7Bs%3A9%3A%22functions%22%3Ba%3A1%3A%7Bs%3A11%3A%22free_result%22%3Bs%3A6%3A%22assert%22%3B%7D%7Ds%3A12%3A%22%00%2A%00recordset%22%3Bs%3A25%3A%22system%28%27cat%20%2Fetc%2Fpasswd%27%29%22%3B%7D"

error log:

    2017/01/12 12:37:34 [info] 32542#32542: *180188 [client 127.0.0.1] ModSecurity: Warning. Matched "Operator `@validatebyterange' with parameter `1-255' against variable `REQUEST_URI' (Value: `/vbulletin/ajax/api/hook/decodeArguments?arguments=O:12:"vB_dB_Result":2:{s:5:"\x00*\x00db";O:17:"vB (135 characters omitted)' ) " at REQUEST_URI [file "/etc/nginx/modsec/owasp-v3/rules/REQUEST-920-PROTOCOL-ENFORCEMENT.conf"] [line "490"] [id "920270"] [rev "2"] [msg "Invalid character in request (null character)"] [data ""] [severity "0"] [ver "OWASP_CRS/3.0.0"] [maturity "9"] [accuracy "9"] [hostname "127.0.0.1"] [uri "/vbulletin/ajax/api/hook/decodeArguments"] [unique_id "14842246542.489138"], client: 127.0.0.1, server: , request: "GET /vbulletin/ajax/api/hook/decodeArguments?arguments=O%3A12%3A%22vB_dB_Result%22%3A2%3A%7Bs%3A5%3A%22%00%2A%00db%22%3BO%3A17%3A%22vB_Database_MySQL%22%3A1%3A%7Bs%3A9%3A%22functions%22%3Ba%3A1%3A%7Bs%3A11%3A%22free_result%22%3Bs%3A6%3A%22assert%22%3B%7D%7Ds%3A12%3A%22%00%2A%00recordset%22%3Bs%3A25%3A%22system%28%27cat%20%2Fetc%2Fpasswd%27%29%22%3B%7D HTTP/1.1", host: "localhost"
    2017/01/12 12:37:34 [info] 32542#32542: *180188 [client 127.0.0.1] ModSecurity: Warning. Matched "Operator `@validatebyterange' with parameter `1-255' against variable `ARGS:arguments' (Value: `O:12:"vB_dB_Result":2:{s:5:"\x00*\x00db";O:17:"vB_Database_MySQL":1:{s:9:"functions";a:1:{s:11:"free (84 characters omitted)' ) " at ARGS:arguments [file "/etc/nginx/modsec/owasp-v3/rules/REQUEST-920-PROTOCOL-ENFORCEMENT.conf"] [line "490"] [id "920270"] [rev "2"] [msg "Invalid character in request (null character)"] [data ""] [severity "0"] [ver "OWASP_CRS/3.0.0"] [maturity "9"] [accuracy "9"] [hostname "127.0.0.1"] [uri "/vbulletin/ajax/api/hook/decodeArguments"] [unique_id "14842246542.489138"], client: 127.0.0.1, server: , request: "GET /vbulletin/ajax/api/hook/decodeArguments?arguments=O%3A12%3A%22vB_dB_Result%22%3A2%3A%7Bs%3A5%3A%22%00%2A%00db%22%3BO%3A17%3A%22vB_Database_MySQL%22%3A1%3A%7Bs%3A9%3A%22functions%22%3Ba%3A1%3A%7Bs%3A11%3A%22free_result%22%3Bs%3A6%3A%22assert%22%3B%7D%7Ds%3A12%3A%22%00%2A%00recordset%22%3Bs%3A25%3A%22system%28%27cat%20%2Fetc%2Fpasswd%27%29%22%3B%7D HTTP/1.1", host: "localhost"
    2017/01/12 12:37:34 [info] 32542#32542: *180188 [client 127.0.0.1] ModSecurity: Warning. Matched "Operator `@pmf' with parameter `unix-shell.data' against variable `ARGS:arguments' (Value: `O:12:"vB_dB_Result":2:{s:5:"\x00*\x00db";O:17:"vB_Database_MySQL":1:{s:9:"functions";a:1:{s:11:"free (84 characters omitted)' ) " at ARGS:arguments [file "/etc/nginx/modsec/owasp-v3/rules/REQUEST-932-APPLICATION-ATTACK-RCE.conf"] [line "412"] [id "932160"] [rev "1"] [msg "Remote Command Execution: Unix Shell Code Found"] [data ""] [severity "0"] [ver "OWASP_CRS/3.0.0"] [maturity "1"] [accuracy "8"] [hostname "127.0.0.1"] [uri "/vbulletin/ajax/api/hook/decodeArguments"] [unique_id "14842246542.489138"], client: 127.0.0.1, server: , request: "GET /vbulletin/ajax/api/hook/decodeArguments?arguments=O%3A12%3A%22vB_dB_Result%22%3A2%3A%7Bs%3A5%3A%22%00%2A%00db%22%3BO%3A17%3A%22vB_Database_MySQL%22%3A1%3A%7Bs%3A9%3A%22functions%22%3Ba%3A1%3A%7Bs%3A11%3A%22free_result%22%3Bs%3A6%3A%22assert%22%3B%7D%7Ds%3A12%3A%22%00%2A%00recordset%22%3Bs%3A25%3A%22system%28%27cat%20%2Fetc%2Fpasswd%27%29%22%3B%7D HTTP/1.1", host: "localhost"
    terminate called after throwing an instance of 'std::out_of_range'
      what():  basic_string::basic_string: __pos (which is 144) > this->size() (which is 28)
    2017/01/12 12:37:34 [notice] 25780#25780: signal 17 (SIGCHLD) received
    2017/01/12 12:37:34 [alert] 25780#25780: worker process 32542 exited on signal 6 (core dumped)

backtrace:

    (gdb) bt
    #0  0x00007f18dc6757ef in raise () from /lib/x86_64-linux-gnu/libc.so.6
    #1  0x00007f18dc6773ea in abort () from /lib/x86_64-linux-gnu/libc.so.6
    #2  0x00007f18db13f58d in __gnu_cxx::__verbose_terminate_handler () at ../../../../src/libstdc++-v3/libsupc++/vterminate.cc:95
    #3  0x00007f18db13d336 in __cxxabiv1::__terminate (handler=<optimized out>) at ../../../../src/libstdc++-v3/libsupc++/eh_terminate.cc:47
    #4  0x00007f18db13d381 in std::terminate () at ../../../../src/libstdc++-v3/libsupc++/eh_terminate.cc:57
    #5  0x00007f18db13d599 in __cxxabiv1::__cxa_throw (obj=obj@entry=0x55ad4b83d160, tinfo=0x7f18db428a98 <typeinfo for std::out_of_range>, dest=0x7f18db153000 <std::out_of_range::~out_of_range()>)
        at ../../../../src/libstdc++-v3/libsupc++/eh_throw.cc:87
    #6  0x00007f18db166ff7 in std::__throw_out_of_range_fmt (__fmt=__fmt@entry=0x7f18dc1c6460 "%s: __pos (which is %zu) > this->size() (which is %zu)")
        at ../../../../../src/libstdc++-v3/src/c++11/functexcept.cc:101
    #7  0x00007f18dc1b4168 in std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_M_check (__s=0x7f18dc1c6443 "basic_string::basic_string", __pos=144, this=0x7fffc2491f80)
        at /usr/include/c++/6/bits/basic_string.h:262
    #8  std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::basic_string (__n=25, __pos=144, __str="O:12:\"vB_dB_Result\":2:{s:5:\"", this=0x7fffc2491fa0)
        at /usr/include/c++/6/bits/basic_string.h:415
    #9  modsecurity::Utils::Regex::searchAll (this=<optimized out>, s=...) at utils/regex.cc:89
    #10 0x00007f18dc1a7028 in modsecurity::operators::Rx::evaluate (this=<optimized out>, transaction=0x55ad4a3fd0e0, rule=0x55ad4a93e810, input=...) at operators/rx.cc:39
    #11 0x00007f18dc19b73a in modsecurity::operators::Operator::evaluateInternal (this=0x55ad4a92f350, transaction=transaction@entry=0x55ad4a3fd0e0, rule=rule@entry=0x55ad4a93e810, 
        a="O:12:\"vB_dB_Result\":2:{s:5:\"\000*\000db\";O:17:\"vB_Database_MySQL\":1:{s:9:\"functions\";a:1:{s:11:\"free_result\";s:6:\"assert\";}}s:12:\"\000*\000recordset\";s:25:\"system('cat /etc/passwd')\";}")
        at operators/operator.cc:80
    #12 0x00007f18dc161961 in modsecurity::Rule::executeOperatorAt (this=this@entry=0x55ad4a93e810, trasn=trasn@entry=0x55ad4a3fd0e0, key="ARGS:arguments", 
        value="O:12:\"vB_dB_Result\":2:{s:5:\"\000*\000db\";O:17:\"vB_Database_MySQL\":1:{s:9:\"functions\";a:1:{s:11:\"free_result\";s:6:\"assert\";}}s:12:\"\000*\000recordset\";s:25:\"system('cat /etc/passwd')\";}") at rule.cc:270
    #13 0x00007f18dc166f62 in modsecurity::Rule::evaluate (this=this@entry=0x55ad4a93e810, trasn=trasn@entry=0x55ad4a3fd0e0) at rule.cc:597
    #14 0x00007f18dc15c64b in modsecurity::Rules::evaluate (this=0x55ad49a626f0, phase=phase@entry=3, transaction=transaction@entry=0x55ad4a3fd0e0) at rules.cc:231
    #15 0x00007f18dc14eea3 in modsecurity::Transaction::processRequestBody (this=0x55ad4a3fd0e0) at transaction.cc:792
    #16 0x00007f18dc150195 in modsecurity::msc_process_request_body (transaction=<optimized out>) at transaction.cc:1779
    #17 0x00007f18dc43cb7e in ngx_http_modsecurity_pre_access_handler (r=0x55ad4b4ea3f0) at ../ModSecurity-nginx//src/ngx_http_modsecurity_pre_access.c:199
    #18 0x000055ad4828821f in ngx_http_core_generic_phase ()
    #19 0x000055ad482838b5 in ngx_http_core_run_phases ()
    #20 0x000055ad4828ea05 in ngx_http_process_request ()
    ...

**Comment by md2k** [comment]
Not sure if i have absolutely same, but my Nginx's workers also dies by Out of range error, as soon as i disabling ModSec Nginx not have any issues.
```
terminate called after throwing an instance of 'std::out_of_range'
  what():  basic_string::basic_string
2017/01/12 22:25:09 [notice] 21649#21649: signal 17 (SIGCHLD) received
2017/01/12 22:25:09 [alert] 21649#21649: worker process 21652 exited on signal 6
2017/01/12 22:25:09 [notice] 21649#21649: start worker process 21677
2017/01/12 22:25:09 [notice] 21649#21649: signal 29 (SIGIO) received
```

**Comment by md2k** [comment]
Seems it is same issue, in gdb i can see also errors with Regex::SearchAll, gdb output for process:

```
(gdb) bt full
#0  0x00007f2921bc9c37 in __GI_raise (sig=sig@entry=6) at ../nptl/sysdeps/unix/sysv/linux/raise.c:56
        resultvar = 0
        pid = 16544
        selftid = 16544
#1  0x00007f2921bcd028 in __GI_abort () at abort.c:89
        save_stage = 2
        act = {__sigaction_handler = {sa_handler = 0x7f2921f521c0 <_IO_2_1_stderr_>, sa_sigaction = 0x7f2921f521c0 <_IO_2_1_stderr_>}, sa_mask = {__val = {139814636666627, 139814613098040, 139814636666627, 139814640099776, 139814640099776, 1, 139814640099907, 140731486982752,
              139814636671964, 1, 139814640099776, 10, 56675232, 140731486982752, 139814636672947, 139814640099776}}, sa_flags = 56674456, sa_restorer = 0x7f292053096c <__cxa_get_globals_fast+12>}
        sigs = {__val = {32, 0 <repeats 15 times>}}
#2  0x00007f2920533535 in __gnu_cxx::__verbose_terminate_handler() () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
No symbol table info available.
#3  0x00007f29205316d6 in ?? () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
No symbol table info available.
#4  0x00007f2920531703 in std::terminate() () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
No symbol table info available.
#5  0x00007f2920531922 in __cxa_throw () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
No symbol table info available.
#6  0x00007f2920583447 in std::__throw_out_of_range(char const*) () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
No symbol table info available.
#7  0x00007f292058e891 in std::basic_string<char, std::char_traits<char>, std::allocator<char> >::basic_string(std::string const&, unsigned long, unsigned long) () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
No symbol table info available.
#8  0x00007f2921707a1b in modsecurity::Utils::Regex::searchAll (this=<optimized out>, s=...) at utils/regex.cc:89
        match = {size_ = 0, match = ""}
        substring_length = 2
        retList = empty std::list
        i = <optimized out>
        ovector = {19817, 19819, 560569040, 32553, 909128241, 3684149, -1706405184, 32766, -1, -1, 0, 23552, 545088472, 32553, 569710432, 32553, 185, 0, 82, 0, 80, 0, -1706404896, 32766, -1706404864, 32766, 566318784, 32553, 64717360, 0}
        rc = 0
#9  0x00007f29216fb1fe in modsecurity::operators::Rx::evaluate (this=<optimized out>, transaction=0x34d03e0, rule=0x314f670, input=...) at operators/rx.cc:39
        match = {size_ = 0, match = <error reading variable: Cannot access memory at address 0xffffffffffffffe8>}
        matches = empty std::list
#10 0x00007f29216eff7a in modsecurity::operators::Operator::evaluateInternal (this=0x314e6b0, transaction=transaction@entry=0x34d03e0, rule=rule@entry=0x314f670,
    a="\354\275\353r\333H\262\060\370[\216\070\357P\246\247Mj\232\067\335,[\022\345ϖ\373\342o\355\266\217\245\236\336\t۫\000\t\220\204\r\002l\000\024\255q+b_c\377\235g9\217\262O\262\231Yw\\H\220\242\334=\033\207\323c\001\250\252\254\254\312KeUee\235\334\177\361\346\354\342\237o\177`\343t\022\234\336;\271\337j\275\367\207\354\345\017\354\361Gxů\354\313$\b\223^m\234\246ӣNg>\237\267\347{\355(\036uv\236<y\322\371\202yjl\020\070\td\362\275\307l>m\245Q\024\364\235\270\306X\340\204\243^\315\v[\277\236\327\020\376{/t\375\341\307VKWv\277\201\325m\263\217\247\370e\235"...) at operators/operator.cc:80
        res = <optimized out>
#11 0x00007f29216b24ac in modsecurity::Rule::executeOperatorAt (this=this@entry=0x314f670, trasn=trasn@entry=0x34d03e0, key="RESPONSE_BODY",
    value="\354\275\353r\333H\262\060\370[\216\070\357P\246\247Mj\232\067\335,[\022\345ϖ\373\342o\355\266\217\245\236\336\t۫\000\t\220\204\r\002l\000\024\255q+b_c\377\235g9\217\262O\262\231Yw\\H\220\242\334=\033\207\323c\001\250\252\254\254\312KeUee\235\334\177\361\346\354\342\237o\177`\343t\022\234\336;\271\337j\275\367\207\354\345\017\354\361Gxů\354\313$\b\223^m\234\246ӣNg>\237\267\347{\355(\036uv\236<y\322\371\202yjl\020\070\td\362\275\307l>m\245Q\024\364\235\270\306X\340\204\243^\315\v[\277\236\327\020\376{/t\375\341\307VKWv\277\201\325m\263\217\247\370e\235"...) at rule.cc:270
        ret = <optimized out>
#12 0x00007f29216b79e4 in modsecurity::Rule::evaluate (this=this@entry=0x314f670, trasn=trasn@entry=0x34d03e0) at rule.cc:597
        ret = <optimized out>
        valueTemp = 0x349f650
        __for_range = <synthetic pointer>
        value = "\354\275\353r\333H\262\060\370[\216\070\357P\246\247Mj\232\067\335,[\022\345ϖ\373\342o\355\266\217\245\236\336\t۫\000\t\220\204\r\002l\000\024\255q+b_c\377\235g9\217\262O\262\231Yw\\H\220\242\334=\033\207\323c\001\250\252\254\254\312KeUee\235\334\177\361\346\354\342\237o\177`\343t\022\234\336;\271\337j\275\367\207\354\345\017\354\361Gxů\354\313$\b\223^m\234\246ӣNg>\237\267\347{\355(\036uv\236<y\322\371\202yjl\020\070\td\362\275\307l>m\245Q\024\364\235\270\306X\340\204\243^\315\v[\277\236\327\020\376{/t\375\341\307VKWv\277\201\325m\263\217\247\370e\235"...
        key = "RESPONSE_BODY"
        values = std::vector of length -6909202, capacity -6909202 = {0x349f650, 0x34f2600, 0x0, 0xd1, 0x30392e322e314147, 0x2e33343930313838, 0x3630383134383431, 0x3003131, 0x7f2921f53434 <initial_quick+404>, 0x34b7e60, 0x0, 0x0, 0x4520293032323134, 0x676e697475636578,
          0x6f74617265706f20, 0x2022787240222072, 0x7261702068746977, 0x3a693f2822206d61, 0x3f782326287c7628, 0x7c29363828282a30, 0x3131287c29363528, 0x29293637287c2938, 0x745c5b2829293f3b, 0x7823282826287c5d, 0x31287c39282a303f, 0x7c293031287c2933, 0x7c293f3b29447c41,
          0x287c293b62617428, 0x3b656e696c77656e, 0x91, 0x69747465732d7077, 0x656d69742d73676e, 0x614b5a44ff00312d, 0x3834317c6b32646d, 0x7c30323334393334, 0x6933706b614b5a44, 0x6f72665357594a63, 0x387331636b4f5a68, 0x45744d6c61475a43, 0x45385a4b79637753, 0x336539657c427842,
          0x6662646436386337, 0x3861636133313034, 0x3535653063666200, 0x3230303662313264, 0x61, 0x34b6080, 0x31, 0x61675f, 0xa, 0x36303831ffffffff, 0x74616d69746c756d, 0x6863, 0x21, 0x7f2921f50032 <_IO_strn_jumps+82>, 0x344e600, 0x20, 0x3a1, 0x314337256b32646d,
          0x3233343933343834, 0x614b5a4443372530, 0x57594a636933706b, 0x6b4f5a686f726653, 0x61475a4338733163, 0x7963775345744d6c, 0x2542784245385a4b, 0x3865303231304337, 0x6565623461343766, 0x6131363535356466, 0x6638366337333936, 0x6232313363626238, 0x3334666136663566,
          0x6161633462383834, 0x6436643536633533, 0x7f2921006533, 0x7f2921f517b8 <main_arena+88>, 0x0, 0x0, 0x353031287c293934, 0x3b29293936287c29, 0x5d745c5b2829293f, 0x3f7823282826287c, 0x3331287c39282a30, 0x417c293031287c29, 0x287c293f3b29447c, 0x6e287c293b626174,
          0x293b656e696c7765, 0x287c70282a292929, 0x28282a303f782326, 0x293035287c293038, 0x287c29323131287c, 0x29293f3b29293037, 0x26287c5d745c5b28, 0x282a303f78232828, 0x287c293331287c39, 0x29447c417c293031, 0x626174287c293f3b, 0x6c77656e287c293b, 0x292929293b656e69,
          0x782326287c74282a, 0x29343828282a303f, 0x31287c293435287c, 0x293437287c293631, 0x5c5b2829293f3b29, 0x23282826287c5d74, 0x287c39282a303f78, 0x293031287c293331, 0x293f3b29447c417c, 0x7c293b626174287c, 0x656e696c77656e28, 0x3a282a292929293b, 0x3f7823282826287c,
          0x7c29383528282a30, 0x293f3b2929413328, 0x3b6e6f6c6f63287c, 0x22292e29292929, 0x5c3030785c303078, 0x341, 0x7f2921f51ae8 <main_arena+904>, 0x7f2921f51ae8 <main_arena+904>, 0x5c3030785c303078, 0x5c3030785c303078, 0x5c3030785c303078, 0x5c3030785c303078,
          0x5c3030785c303078, 0x5c3030785c303078, 0x5c3030785c303078, 0x2f1, 0x7f2921f517b8 <main_arena+88>, 0x7f2921f517b8 <main_arena+88>, 0x5c3030785c303078, 0x5c3030785c303078, 0x5c3030785c303078, 0x5c3030785c303078, 0x5c3030785c303078, 0x5c3030785c303078,
          0x5c3030785c303078, 0x5c3030785c303078, 0x5c3030785c303078, 0x5c3030785c303078, 0x5c3030785c303078, 0x5c3030785c303078, 0x5c3030785c303078, 0x5c3030785c303078, 0x5c3030785c303078, 0x5c3030785c303078, 0x5c3030785c303078, 0x5c3030785c303078, 0x5c3030785c303078,
          0x5c3030785c303078, 0x5c3030785c303078, 0x5c3030785c303078, 0x5c3030785c303078, 0x5c3030785c303078, 0x5c3030785c303078, 0x5c3030785c303078, 0x5c3030785c303078, 0x5c3030785c303078, 0x5c3030785c303078, 0x5c3030785c303078, 0x5c3030785c303078, 0x5c3030785c303078,
          0x5c3030785c303078, 0x5c3030785c303078, 0x5c3030785c303078, 0x5c3030785c303078, 0x5c3030785c303078, 0x5c3030785c303078, 0x5c3030785c303078, 0x5c3030785c303078, 0x5c3030785c303078, 0x5c3030785c303078, 0x5c3030785c303078, 0x181, 0x7365727064726f77, 0x646567676f6c5f73,
          0x646365325f6e695f, 0x3731363164343832, 0x6363386130343065, 0x6635633734333961, 0x2d656d0035373939, 0x6972615628202231, 0x4552203a656c6261, 0x4f435f5453455551, 0x414e5f5345494b4f, 0x732d70773a53454d, 0x2d73676e69747465, 0x29312d656d6974, 0x3230303662313264, 0x101,
          0x34223f0, 0x34b2570, 0x22296400ffffffff, 0x3528205420, 0x7f2921f517b8 <main_arena+88>, 0xd1, 0x7f2921f51878 <main_arena+280>, 0xc1, 0x7f2921f517b8 <main_arena+88>, 0x7f2921f517b8 <main_arena+88>...}
        multiMatch = <optimized out>
        v = <optimized out>
        __for_range = <synthetic pointer>
        variables = <optimized out>
        recursiveGlobalRet = <optimized out>
        containsDisruptive = false
        ruleMessage = {m_match = "", m_ruleFile = "/etc/nginx/modsec/owasp/rules/RESPONSE-953-DATA-LEAKAGES-PHP.conf", m_ruleLine = 81, m_ruleId = 953120, m_message = "", m_data = "", m_severity = 0, m_ver = "OWASP_CRS/3.0.0", m_rev = "2", m_maturity = 9, m_accuracy = 9,
          m_tags = empty std::list, m_server_logs = empty std::list, m_noAuditLog = false, m_rule = 0x314f670, m_saveMessage = false}
        eparam = "\"<\\?(?!xml)\""
        globalRet = false
---Type <return> to continue, or q <return> to quit---
        finalVars = std::vector of length 1, capacity -6927956 = {0x34f1660}
#13 0x00007f29216ad1e3 in modsecurity::Rules::evaluate (this=0x27395c0, phase=phase@entry=5, transaction=transaction@entry=0x34d03e0) at rules.cc:231
        rule = 0x314f670
        i = 63
        rules = <optimized out>
#14 0x00007f29216a3efb in modsecurity::Transaction::processResponseBody (this=0x34d03e0) at transaction.cc:1080
        bi = <optimized out>
        t = <optimized out>
#15 0x00007f29216a4265 in modsecurity::msc_process_response_body (transaction=<optimized out>) at transaction.cc:1863
No locals.
#16 0x00007f292199026e in ngx_http_modsecurity_body_filter (r=0x339ee10, in=0x3777470) at ../ModSecurity-nginx//src/ngx_http_modsecurity_body_filter.c:164
        ret = <optimized out>
        old_pool = 0x0
        buffer_fully_loadead = <optimized out>
        chain = <optimized out>
        ctx = 0x33b4550
```

**Comment by zimmerle** [comment]
Fixed at: 36d6bb96645cb5ab085417089de2ccce7825cdb1.
